framework/test/unit/engine/endpoint/chain.spec.ts

```diff
@@ -229,4 +280,254 @@ describe('Chain endpoint', () => {
+			).resolves.toEqual(transaction.toJSON());
+		});
+	});
+
+	describe('getTransactionsByIDs', () => {
+		it('should throw if provided ids is empty or not an array', async () => {
+			await expect(
+				endpoint.getTransactionsByIDs(createRequestContext({ ids: [] })),
+			).rejects.toThrow('Invalid parameters. ids must be a non empty array');
+
+			await expect(
+				endpoint.getTransactionsByIDs(createRequestContext({ ids: 'invalid id' })),
+			).rejects.toThrow('Invalid parameters. ids must be a non empty array');
+		});
+
+		it('should throw if any of the provided ids are not valid', async () => {
+			await expect(
+				endpoint.getTransactionsByIDs(createRequestContext({ ids: [validBlockID, 'invalid ID'] })),
+			).rejects.toThrow('Invalid parameters. id must be a valid hex string.');
+		});
+
+		it('should return a collection of transactions', async () => {
+			jest
+				.spyOn(endpoint['_chain'].dataAccess, 'getTransactionByID')
+				.mockResolvedValue(transaction);
+
+			await expect(
+				endpoint.getTransactionsByIDs(
+					createRequestContext({ ids: [transaction.id.toString('hex')] }),
+				),
+			).resolves.toEqual([transaction.toJSON()]);
+		});
+	});
+
+	describe('getTransactionsByHeight', () => {
+		it('should throw if provided height is invalid', async () => {
+			await expect(
+				endpoint.getTransactionsByHeight(createRequestContext({ height: 'invalid height' })),
+			).rejects.toThrow('Invalid parameters. height must be zero or a positive number.');
+
+			await expect(
+				endpoint.getTransactionsByHeight(createRequestContext({ height: -1 })),
+			).rejects.toThrow('Invalid parameters. height must be zero or a positive number.');
+		});
+
+		it('should return a collection of transactions in the block at the provided height', async () => {
+			jest.spyOn(endpoint['_chain'].dataAccess, 'getBlockByHeight').mockResolvedValue(block);
+
+			await expect(
+				endpoint.getTransactionsByHeight(createRequestContext({ height: 1 })),
+			).resolves.toEqual(block.transactions.map(t => t.toJSON()));
+		});
+	});
+
+	describe('getAssetsByHeight', () => {
+		it('should throw if provided height is invalid', async () => {
+			await expect(
+				endpoint.getAssetsByHeight(createRequestContext({ height: 'invalid height' })),
+			).rejects.toThrow('Invalid parameters. height must be zero or a positive number.');
+
+			await expect(
+				endpoint.getAssetsByHeight(createRequestContext({ height: -1 })),
+			).rejects.toThrow('Invalid parameters. height must be zero or a positive number.');
+		});
+
+		it('should return block assests at the provided height', async () => {
+			jest.spyOn(endpoint['_chain'].dataAccess, 'getBlockByHeight').mockResolvedValue(block);
+
+			await expect(
+				endpoint.getAssetsByHeight(createRequestContext({ height: 1 })),
+			).resolves.toEqual(block.assets.toJSON());
+		});
+	});
+
+	describe('areHeadersContradicting', () => {
+		it('should if provided parameters are not valid', async () => {
```

Comment:
This desc is not so clear..maybe it's just me 🙃. Should do what exactly?

Reply:
This is definitely unclear 😅, I will update.

Reply:
Besides this, LGTM! 🚀